test/e2e_mc/multicluster_test.go

```diff
@@ -0,0 +1,154 @@
+	westClusterTD := testData.clusterTestDataMap[westCluster]
+	if err := data.createPodOnNode(eastClusterTD, "perftest-a", multiClusterTestNamespace, data.controlPlaneNames[eastCluster], perftoolImage, nil, nil, nil, nil, false, nil); err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("error when creating the perftest client Pod: %v", err)
+	}
+	podAIPs, err = data.podWaitForIPs(defaultTimeout, eastCluster, "perftest-a", multiClusterTestNamespace)
+	if err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("error when waiting for the perftest client Pod: %v", err)
+	}
+	if err := data.createPodOnNode(eastClusterTD, "perftest-b", multiClusterTestNamespace, data.controlPlaneNames[eastCluster], perftoolImage, nil, nil, nil, []corev1.ContainerPort{{Protocol: corev1.ProtocolTCP, ContainerPort: iperfPort}}, false, nil); err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("error when creating the perftest server Pod: %v", err)
+	}
+	podBIPs, err = data.podWaitForIPs(defaultTimeout, eastCluster, "perftest-b", multiClusterTestNamespace)
+	if err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("error when getting the perftest server Pod's IPs: %v", err)
+	}
+	if err := data.createPodOnNode(westClusterTD, "perftest-c", multiClusterTestNamespace, data.controlPlaneNames[westCluster], perftoolImage, nil, nil, nil, nil, false, nil); err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("error when creating the perftest server Pod: %v", err)
+	}
+	podCIPs, err = data.podWaitForIPs(defaultTimeout, westCluster, "perftest-c", multiClusterTestNamespace)
+	if err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("error when getting the perftest server Pod's IPs: %v", err)
+	}
+	if err := data.createPodOnNode(westClusterTD, "perftest-d", multiClusterTestNamespace, data.controlPlaneNames[westCluster], perftoolImage, nil, nil, nil, []corev1.ContainerPort{{Protocol: corev1.ProtocolTCP, ContainerPort: iperfPort}}, false, nil); err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("error when creating the perftest server Pod: %v", err)
+	}
+	podDIPs, err = data.podWaitForIPs(defaultTimeout, westCluster, "perftest-d", multiClusterTestNamespace)
+	if err != nil {
+		return nil, nil, nil, nil, fmt.Errorf("error when getting the perftest server Pod's IPs: %v", err)
+	}
+	return podAIPs, podBIPs, podCIPs, podDIPs, nil
+}
+
+func failOnError(err error, t *testing.T, data *MCTestData) {
+	flowVisibilityCleanup(t, data)
+	t.Fatalf("test failed: %v", err)
+}
+
+func TestMCCoverageAndCleanup(t *testing.T) {
+	if os.Getenv("COVERAGE") == "" {
+		t.Skipf("COVERAGE env variable is not set until all tests are run, skipping test.")
+	}
+	workerNodeNames := []string{"east-worker", "west-worker"}
+	covDirs := []string{".coverage/clickhouse-monitor-coverage", ".coverage/theia-manager-coverage"}
+	covDirPrefixes := []string{"cm", "tm"}
+	tearDownBothNodes(workerNodeNames, covDirs, covDirPrefixes)
+}
+
+func tearDownBothNodes(workerNodeNames, covDirs, covDirPrefixes []string) error {
```

Comment:
I feel `tearDownBothNodes` might not directly related to this specific test and could potentially be moved to `framework.go`.
Maybe `TestMCCoverageAndCleanup` as well, I don't have a strong opinion on it.

Reply:
I've renamed `tearDownBothNodes ` to `collectCovFromNodes` and put `TestMCCoverageAndCleanup ` inside `flowVisibilityCleanup`.